src/app/modules/main/profile_section/profile/models/profile_save_data.nim

```diff
@@ -0,0 +1,50 @@
+import json, strutils, sequtils
+
+include app_service/common/json_utils
+include app_service/common/utils
+
+import app_service/service/profile/dto/profile_showcase_preferences
+
+type ProfileShowcaseSaveEntry* = ref object of RootObj
```

Comment:
Why do we need a different type than `ProfileShowcasePreferencesItem` in here?

Reply:
In case we have similar requirements to the model source item and json structure, it may work. But on the latest discussion stage they are a bit different.

Reply:
Which are the differences?

Reply:
fields naming and extra fields for collectibles

Reply:
Even if they are two identical structures, I'm against using the same type for models and for reflecting the json structure. It would be a time bomb.

Reply:
> Even if they are two identical structures, I'm against using the same type for models and for reflecting the json structure. It would be a time bomb.

Probably I'm missing something but I'd opt for using same type of object when we are talking about the same data instead of creating different types each time we do different actions, otherwise we are acumulating types and types that indeed define the same things. However, you have a better view of the complete flow so I leave it up to you! It was just a suggestion! :)